### PR TITLE
email-job

### PR DESCRIPTION
**Pull request by t-harps**
adding unsubscription link to emails, subscription attribute to user.

Emails now contain an unsubscribe link generated by [MessageVerifier](http://api.rubyonrails.org/classes/ActiveSupport/MessageVerifier.html). This link takes them to a page where they can unsubscribe from emails without having to log in. 

Followed [this guide](https://stackoverflow.com/a/34779543) as mentioned in redmine ticket.

### CODE REVIEW

**Comment by dmitrytrager** [comment]
Looks like there is a special route for user to unsibscribe.
@t-harps could you please describe the whole flow in PR description?

**Comment by t-harps** [comment]
@dmitrytrager Added to description plus put link to stack overflow answer i used.

File: spec/controllers/settings_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  describe "GET #unsubscribe" do
+    it "returns http success" do
+      get :unsubscribe, params: { id: unsub_id }
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe "PUT #update" do
+    context "with subscription = false" do
```
Comment (dmitrytrager) [inline spec/controllers/settings_controller_spec.rb]: probably, 'with subscription set to false'

File: spec/controllers/settings_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+require "rails_helper"
+
+RSpec.describe SettingsController, type: :controller do
+  let(:user) { create :user }
+  let(:unsub_id) { Rails.application.message_verifier(:unsubscribe).generate(user.id) }
+  let(:unsubscribe) { put :update, params: { id: user.id, user: { subscription: false } } }
```
Comment (dmitrytrager) [inline spec/controllers/settings_controller_spec.rb]: method is also possible for DRY here

**Pull request merged**